rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -577,6 +594,14 @@ public void internalAddNodeToCore(RMNode configuredNode) {
         internalSetFree(configuredNode);
     }
 
+    private boolean isNumberOfNodesLimited() {
+        return maximumNumberOfNodes != null && maximumNumberOfNodes > 0;
```

Comment:
zero should be a legal value, so I would write the test rather:

return maximumNumberOfNodes != null && maximumNumberOfNodes >= 0;